src/app/actions/admin/products.ts

```diff
@@ -3,22 +3,37 @@ import { AvailabilityStatus, Category, returnPolicy } from '@/types/product';
 import { z } from 'zod';
 import { db, collections } from '@/utils/firebase';
 import { doc, setDoc } from "firebase/firestore";
+import { put } from '@vercel/blob';
+
 
 export const productSchema = z.object({
```

Comment:
You normally should be able to use the same schema for creating and updating the product, it would make your solution much cleaner and reliable. Image could be present but you can normally skip ID (since the user does not need to pass it from the client side) or at least make it optional.